src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -363,9 +370,11 @@ module.exports = function(
       // If there aren't any new props, then we'll reuse the memoized props.
       // This could be from already completed work.
       newProps = oldProps;
-      if (!newProps) {
-        throw new Error('There should always be pending or memoized props.');
-      }
+      invariant(
+        newProps !== null,
```

Comment:
Yeah I'll probably loosen those up. Should do a pass at some point and change all our maybe types to null unions. Though props is currently an `any`, which is worse.